packages/client/src/monaco-language-client.ts

```diff
@@ -2,81 +2,83 @@
  * Copyright (c) 2018-2022 TypeFox GmbH (http://www.typefox.io). All rights reserved.
  * Licensed under the MIT License. See License.txt in the project root for license information.
  * ------------------------------------------------------------------------------------------ */
+
+/* eslint-disable dot-notation */
```

Comment:
Are you sure it's a good idea to disable a rule for a whole file?

Reply:
No, it should not be there. I changed the config to this now (upcoming commit)
```javascript
        'dot-notation': 'off',
        '@typescript-eslint/dot-notation': ['error']
```
but now VSCode is auto-formatting the code to dotted notation all the time and this leads to TypeScript errors, for example:

```shell
src/monaco-language-client.ts:55:103 - error TS2341: Property 'handleRegistrationRequest' is private and only accessible within class 'BaseLanguageClient'.

55         const originalHandleRegistrationRequest: (params: RegistrationParams) => Promise<void> = this.handleRegistrationRequest.bind(this);
```

Reply:
Yes, this is a known hack to access a private field in typescript!

The best way is to disable the dot-notation error on each line I think

Reply:
Did this now. See new commit